crates/bevy_ui/src/layout/mod.rs

```diff
@@ -158,6 +160,82 @@ pub fn ui_layout_system(
         }
     }
 
+    for new_root_node_entity in removed_parents.read() {
+        // TODO: most of this should be moved into a function on `UiSurface` but the query requirements might feel out of place
```

Comment:
I was planning on moving this logic into an associated function on `UiSurface`, but passing the queries seems out of scope. Open to suggestions.